codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -214,9 +218,41 @@ public ChannelFuture sendFlowControlledFrame(ChannelHandlerContext ctx, Http2Str
     }
 
     @Override
-    public ChannelFuture lastFlowControlledFrameSent(Http2Stream stream) {
-        FlowState state = state(stream);
-        return state != null ? state.lastNewFrame() : null;
+    public ChannelFuture sendHeaders(ChannelHandlerContext ctx, Http2Stream stream, Http2Headers headers,
+            int streamDependency, short weight, boolean exclusive, int padding,
+            boolean endOfStream, ChannelPromise promise) {
+      boolean isFirst = !sentFrame;
```

Comment:
I think the convention to verify stream steams lives in `DefaultHttp2ConnectionEncoder` and for headers [here](https://github.com/netty/netty/blob/master/codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java#L232).  Is there a reason this is not sufficient?